synapse/app/_base.py

```diff
@@ -390,44 +390,47 @@ def listen_http(
     context_factory: Optional[IOpenSSLContextFactory],
     reactor: ISynapseReactor = reactor,
 ) -> List[Port]:
-    port = listener_config.port
-    bind_addresses = listener_config.bind_addresses
-    tls = listener_config.tls
-
     assert listener_config.http_options is not None
 
     site_tag = listener_config.http_options.tag
     if site_tag is None:
-        site_tag = str(port)
+        site_tag = listener_config.get_site_tag()
```

Comment:
since `http_options.tag` is part of `listener_config`, I wonder if we can or should put this `if` block in `get_site_tag`? We *do* use it twice..